### PR TITLE
Allow background & camera icon customization, default to 4:3 ratio, don't dismiss picker after taking a photo, and format the code

### PR DESCRIPTION
**Pull request by hyouuu**
Looks like a lot of changes, but they're actually code formation since previously the indentions had lots of issues.

The core changes are:
1. Added a cameraIcon property which will be used in the camera cell
2. Added a overrideBackgroundColor property to customize the picker background color

Thanks for the component!

### CODE REVIEW

**Comment by zhangao0086** [comment]
Could you change 2-space indent to 4-space?

**Comment by hyouuu** [comment]
Updated

File: DKImagePickerController/DKImagePickerController.swift [hunk 1]
```diff
@@ -206,104 +212,105 @@ public class DKImagePickerController: UINavigationController {
...
+            PHPhotoLibrary.sharedPhotoLibrary().performChanges(
+                { () in
+                    let assetRequest = PHAssetChangeRequest.creationRequestForAssetFromImage(image)
+                    newImageIdentifier = assetRequest.placeholderForCreatedAsset!.localIdentifier
+                }, completionHandler: { (success, error) -> Void in
+                    dispatch_async(dispatch_get_main_queue(), { () -> Void in
+                        if success {
+                            if let newAsset = PHAsset.fetchAssetsWithLocalIdentifiers([newImageIdentifier], options: nil).firstObject as? PHAsset {
+                                self.dismissViewControllerAnimated(true, completion: nil)
+                                self.selectedImage(DKAsset(originalAsset: newAsset), needsToDismiss: false)
```
Comment (zhangao0086) [inline DKImagePickerController/DKImagePickerController.swift]: May affect the `Take a picture`?

**Comment by zhangao0086** [comment]
Maybe I need to implement a `UIDelegate` to support custom UI.

File: DKImagePickerController/DKImagePickerController.swift (hunk 1 above)
Comment (hyouuu) [inline DKImagePickerController/DKImagePickerController.swift]: So the difference is that it won't dismiss the gallery picking view directly after capturing. Are you saying there's another mode where taking a picture is the sole usage? In that case we probably should add a condition here

**Comment by hyouuu** [comment]
Not familiar with UIDelegate, but probably we could have these properties then convert to the delegate later?

**Comment by zhangao0086** [comment]
Why're you change this line
```swift
self.selectedImage(DKAsset(originalAsset: newAsset), needsToDismiss: true)
```
to 
```swift
self.selectedImage(DKAsset(originalAsset: newAsset), needsToDismiss: false)
```
?
I think this will cause the `Take a picture` function was broken. If you want to [have an intermediate view for Camera](https://github.com/zhangao0086/DKImagePickerController/issues/83), I think a custom camera is better, in that case the `didCancel` and the `didFinishCapturingImage` can be called when you want.

Then I'll implement a protocol like this:
```swift
public protocol DKImagePickerControllerCameraProtocol {
	
	static func checkCameraPermission(handler: (granted: Bool) -> Void)
	
	var didCancel: (() -> Void)? { get set }
	
	var didFinishCapturingImage: ((image: UIImage) -> Void)? { get set }
	
	var cameraOverlayView: UIView? { get set }
	
}
```

All classes which implement the protocol will be able to use in the picker controller. What do you think?

**Comment by hyouuu** [comment]
Exactly as you said, it was for the intermediate view. I've reverted that part so we can deal with that separately, e.g. through the delegate you mentioned

**Comment by zhangao0086** [comment]
I've made a new branch [Feature/CustomCamera](https://github.com/zhangao0086/DKImagePickerController/tree/Feature/CustomCamera). A custom camera protocol  has been added to the the branch.

You can implement this protocol:
```swift
@objc
public protocol DKImagePickerControllerUIDelegate {
	
	func imagePickerControllerCreateCamera(imagePickerController: DKImagePickerController,
		didCancel: (() -> Void),
		didFinishCapturingImage: ((image: UIImage) -> Void)) -> UIViewController
	
}
```

then set the `UIDelegate` like so:
```swift
pickerController.UIDelegate = CustomUIDelegate()
```

Let me know if you have any questions

**Comment by zhangao0086** [comment]
The following is the default `UIDelegate`:
```swift
@objc
public class DKImagePickerControllerDefaultUIDelegate: NSObject, DKImagePickerControllerUIDelegate {
	
	public func imagePickerControllerCreateCamera(imagePickerController: DKImagePickerController, didCancel: (() -> Void), didFinishCapturingImage: ((image: UIImage) -> Void)) -> UIViewController {
		
		let camera = DKCamera()
		
		camera.didCancel = { () -> Void in
			didCancel()
		}
		
		camera.didFinishCapturingImage = { (image) in
			didFinishCapturingImage(image: image)
		}
		self.checkCameraPermission(camera)
	
		return camera
	}
	
	private func checkCameraPermission(camera: DKCamera) {
		func cameraDenied() {
			dispatch_async(dispatch_get_main_queue()) {
				let permissionView = DKPermissionView.permissionView(.Camera)
				camera.cameraOverlayView = permissionView
			}
		}
		
		func setup() {
			camera.cameraOverlayView = nil
		}
		
		DKCamera.checkCameraPermission { granted in
			granted ? setup() : cameraDenied()
		}
	}
}
```

**Comment by zhangao0086** [comment]
In `3.1.0`, you can use your custom camera instead of DKCamera. Simply implement the `DKImagePickerControllerUIDelegate` protocol or inherit the `DKImagePickerControllerDefaultUIDelegate` class.

**Comment by hyouuu** [comment]
Thanks! There were a few places where I had to change the internal or private to public, but a great start!

**Pull request closed without merging**